src/server/pfs/server/driver.go

```diff
@@ -807,25 +807,25 @@ func (d *driver) listProject(ctx context.Context, cb func(*pfs.ProjectInfo) erro
 	if err != nil {
 		return errors.Wrap(err, "could not list projects")
 	}
-	defer projIter.Close()
-	for projectInfo, err := projIter.Next(); !errors.Is(err, io.EOF); projectInfo, err = projIter.Next() {
+	projInfo := &pfs.ProjectInfo{}
+	for err := projIter.Next(ctx, projInfo); !errors.Is(err, io.EOF); err = projIter.Next(ctx, projInfo) {
```

Comment:
Is GitHub displaying this correctly?? I'm seeing two calls to `.Next()`
```Go
for err := projIter.Next(ctx, projInfo); !errors.Is(err, io.EOF); err = projIter.Next(ctx, projInfo)
```

Reply:
Yes, so the first call gets the first entry and the second call advances the iterator similar to
go ```
for int i := 0; i < 100; i++ {
}
````

Reply:
I find this pattern really weird... doesn't https://github.com/pachyderm/pachyderm/pull/8951/files#diff-fe7ebf0ab8e4af97b3c44edd1b2666f1420b0e08053d1042caf22f1af212dec1R60 automatically advance the iterator?

Reply:
can't you do the same with something like
```Go
for !errors.Is(projectIter.Next(ctx, projInfo), io.EOF) {}
```

Reply:
Maybe! I can try that out. I agree this `for` pattern is awkward. Its more of a holdover from when the project was returned by Next()